File: src/main/java/com/smarthome/backend/model/TriggerConfig.java
package com.smarthome.backend.model;

import lombok.Data;

/**
 * Konfiguration für einen Trigger-Knoten im Workflow.
 */
@Data
public class TriggerConfig {
    private String type; // 'manual' | 'device' | 'time'
    private DeviceTrigger device; // Optional, wenn type 'device'
    private TimeTrigger time; // Optional, wenn type 'time'

    public DeviceTrigger getDeviceTrigger(){
        return device;
    }

    public TimeTrigger getTimeTrigger(){
        return time;
    }
}

File: src/main/node/com/smarthome/backend/model/TriggerConfig.ts
import { DeviceTrigger } from "./DeviceTrigger.js";
import { TimeTrigger } from "./TimeTrigger.js";

export class TriggerConfig {
  type?: string;
  device?: DeviceTrigger;
  time?: TimeTrigger;

  constructor(init?: Partial<TriggerConfig>) {
    Object.assign(this, init);
  }
}
